lib/request_handler/body_parser.rb

```diff
@@ -25,7 +25,7 @@ def run
 
     def flattened_request_body
       body = request_body.fetch('data') do
-        raise ExternalArgumentError, body: 'must contain data'
+        raise BodyParamsError, body: 'must contain data'
```

Comment:
With errors like this we would throw redundant information as the error iself already contains information about what caused it in key value pairs.
This also allows one external error to be caused by multiple issues.

If we add different error classes, we should get rid of the error information inside the error objects and replace them with a simple message.

I think we should keep it the way it is right now as this also allows the gem to be extended more easily.  You could handle the errors different by querying the error(s) inside the error object.  What's your opinion on this @andreaseger ?